--- res/人工智能训练师三级上网素材/2.1.2/2.1.2.ipynb.py
+++ res/人工智能训练师三级上网素材/2.1.2/2.1.2.ans.ipynb.py
@@ -3 +3 @@
-data = __________

+data = pd.read_excel('大学生低碳生活行为的影响因素数据集.xlsx')

@@ -5,0 +6 @@
+print(data.columns.to_list())

@@ -8,3 +9,3 @@
-initial_row_count = __________   #处理前的数据行数

-data = __________                #删除缺失值所在行

-final_row_count = __________     #处理后的数据行数

+initial_row_count = data.shape[0]   #处理前的数据行数

+data = data.dropna()                #删除缺失值所在行

+final_row_count =data.shape[0]    #处理后的数据行数

@@ -14 +15 @@
-data = __________

+data = data.drop_duplicates()

@@ -19 +20 @@
-data[numerical_features] = __________

+data[numerical_features] = scaler.fit_transform(data[numerical_features])

@@ -22,2 +23,17 @@
-selected_features = [__________]

-X = __________

+selected_features = [

+    '1.您的性别○男性   ○女性',

+   '2.您的年级○大一   ○大二   ○大三   ○大四', 

+   '3.您的生源地○农村   ○城镇（乡镇）   ○地县级城市  ○省会城市及直辖市',

+     '4.您的月生活费○≦1,000元   ○1,001-2,000元   ○2,001-3,000元   ○≧3,001元',

+       '5.您进行过绿色低碳的相关生活方式吗?', 

+       '6.您觉得“低碳”，与你的生活关系密切吗？', 

+       '7.低碳生活是否会成为未来的主流生活方式？', 

+       '8.您是否认为低碳生活会提高您的生活质量？'

+    ]

+

+ 

+

+        

+

+

+X = data[selected_features]

@@ -26 +42 @@
-y = __________

+y = data['低碳行为积极性']

@@ -30 +46 @@
-X_train, X_test, y_train, y_test = __________(__________, random_state=42)

+X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=0.2, random_state=42)

@@ -33,2 +49,2 @@
-cleaned_data = __________(__________, axis=1)

-__________('2.1.2_cleaned_data.csv', __________)

+cleaned_data = pd.concat([X,y], axis=1)

+cleaned_data.to_csv('2.1.2_cleaned_data.csv', index=False)

